apps/prairielearn/src/pages/instructorAssessmentAccessOverrides/instructorAssessmentAccessOverrides.sql

```diff
@@ -0,0 +1,105 @@
+-- BLOCK select_assessment_access_policies
+SELECT
+  format_date_full_compact (aap.created_at, 'America/Chicago') AS created_at,
+  aap.created_by::text AS created_by,
+  aap.credit::text AS credit,
+  format_date_full_compact (aap.end_date, 'America/Chicago') AS end_date,
+  aap.note AS note,
+  format_date_full_compact (aap.start_date, 'America/Chicago') AS start_date,
+  (
+    SELECT
+      name
+    from
+      groups
+    where
+      id = aap.group_id
+  ) as group_name,
+  (
+    SELECT
+      uid
+    FROM
+      users
+    WHERE
+      user_id = aap.user_id
+  ) as student_uid,
+  aap.id as id
+FROM
+  assessment_access_policies AS aap
+WHERE
+  assessment_id = $assessment_id
+ORDER BY
+  aap.created_at;
+
+-- BLOCK select_group_in_assessment
+SELECT
+  groups.id,
+  groups.name,
+  groups.group_config_id
+FROM
+  groups
+  JOIN group_configs ON groups.group_config_id = group_configs.id
+WHERE
+  groups.course_instance_id = $course_instance_id
+  AND groups.name = $group_name
+  AND group_configs.assessment_id = $assessment_id;
+
+-- BLOCK insert_assessment_access_policy
+INSERT INTO
+  assessment_access_policies (
+    assessment_id,
+    created_at,
+    created_by,
+    credit,
+    end_date,
+    group_id,
+    note,
+    start_date,
+    user_id
+  )
+VALUES
+  (
+    $assessment_id,
+    NOW(),
+    $created_by,
+    $credit,
+    $end_date,
+    $group_id,
+    $note,
+    $start_date,
+    $user_id
+  );
+
+-- BLOCK update_assessment_access_policy
+UPDATE assessment_access_policies
+SET
+  created_at = NOW(),
+  credit = $credit,
+  end_date = $end_date,
+  group_id = $group_id,
+  note = $note,
+  start_date = $start_date,
+  user_id = (
+    SELECT
+      user_id
+    FROM
+      users
+    WHERE
+      uid = $student_uid
+  )
+WHERE
+  assessment_id = $assessment_id
+  AND (
+    user_id = (
+      SELECT
+        user_id
+      FROM
+        users
+      WHERE
+        uid = $student_uid
+    )
+  );
```

Comment:
This doesn't handle group work. You should pass in an `assessment_access_policies.id` value and use that to target which row to update. As with `delete_assessment_access_policy`, take care to validate that the references access policy actually belongs to the current assessment.

Reply:
Changes made in dbd3543

Reply:
There's no need to shorten the name to `policy_id`; we try to keep full table names as part of IDs for clarity. So, in this case, you'd want to use `assessment_access_policies_id`.

Reply:
561fd1b1c fixes the table name.